src/Microsoft.AspNet.Mvc.Core/Controller.cs

```diff
@@ -376,6 +376,11 @@ public virtual ContentResult Content(string content, string contentType, Encodin
         [NonAction]
         public virtual JsonResult Json(object data)
         {
+            var disposableValue = data as IDisposable;
+            if (disposableValue != null)
+            {
+                Response.OnResponseCompleted(_ => disposableValue.Dispose(), null);
```

Comment:
All exceptions are caught and are logged. 
https://github.com/aspnet/WebListener/pull/92/files#diff-c5fd73459400ce91b25076b492d3d91eR878